### PR TITLE
Fix/3478/reset camera global option broken

### PR DESCRIPTION
**Pull request by IdrisLaabidi**
# Fix camera breaking because of the global option 'reset camera when changing map'

Issue: #3478 

## Description
  - Fix the bug described at #3478 
  - The camera no longer breaks if you reload the page when while the reset camera global option is set to false. Now it will auto fit.

## Definition of Done

A PR is only ready for merge once all the following acceptance criteria are fulfilled:
- [x] Changes have been manually tested
- [x] All TODOs related to this PR have been closed
- [x] There are automated tests for newly written code and bug fixes
- [x] All bugs discovered while working on this PR have been submitted as issues (if not already an open issue)
- [ ] ~Documentation (GH-pages, analysis/visualization READMEs, parser READMEs, --help, etc.) has been updated (almost always necessary except for bug fixes)~
- [x] CHANGELOG.md has been updated

### CODE REVIEW

File: visualization/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 ### Fixed 🐞
 
 - Fix issue where zooming out too much makes the map disappear and zooming in too much causes you to go through the map. [#3697](https://github.com/MaibornWolff/codecharta/pull/3697)
+- Camera perspective is correctly adopted from the custom configuration[#3194](https://github.com/MaibornWolff/codecharta/pull/3194)
```
Comment (IhsenBouallegue) [inline visualization/CHANGELOG.md:18]: Not so sure why, but this redirects to an issue. According to our guidelines it should be a PR, preferably this one.

File: visualization/CHANGELOG.md [hunk 2]
```diff
@@ -29,7 +31,6 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 ### Fixed 🐞
 
 - Revert [#3655](https://github.com/MaibornWolff/codecharta/pull/3665) as we implement new navigation methods
-- Camera perspective is correctly adopted from the custom configuration[#3194](https://github.com/MaibornWolff/codecharta/pull/3194)
```
Comment (IhsenBouallegue) [inline visualization/CHANGELOG.md:32]: Why do we need to change the changelog of previous versions?

File: visualization/app/codeCharta/ui/codeMap/threeViewer/threeMapControls.service.spec.ts [hunk 3]
```diff
@@ -113,6 +136,17 @@ describe("ThreeMapControlsService", () => {
...
+
+        it("should return early if boundingSphere.radius is -1", async () => {
+            jest.spyOn(threeMapControlsService, "getBoundingSphere").mockReturnValue(new Sphere())
+            threeCameraService.camera.position.set(0, 0, 0)
+
+            threeMapControlsService.autoFitTo()
+            await wait(0)
+
+            expect(threeCameraService.camera.position).toEqual(new Vector3(0, 0, 0))
+            //expect(threeRendererService.render).not.toHaveBeenCalled()
```
Comment (IhsenBouallegue) [inline visualization/app/codeCharta/ui/codeMap/threeViewer/threeMapControls.service.spec.ts:148]: remove commented line if not being used

**Review by IhsenBouallegue — APPROVED** [review]
Just minor nitpicks otherwise looks good 👍

File: visualization/CHANGELOG.md (hunk 2 above)
Comment (IdrisLaabidi) [reply visualization/CHANGELOG.md:32]: Actually, that's a recent bug fix, it has nothing to do with the previous version. It was there by mistake, and I corrected it

**Pull request merged**